src/librustc/ty/wf.rs

```diff
@@ -202,11 +201,10 @@ impl<'a, 'gcx, 'tcx> WfPredicates<'a, 'gcx, 'tcx> {
         }
     }
 
-    /// Pushes the obligations required for a constant value to be WF
+    /// Pushes the obligations required for an array length to be WF
     /// into `self.out`.
-    fn compute_const(&mut self, constant: &'tcx ty::Const<'tcx>) {
-        self.require_sized(constant.ty, traits::ConstSized);
-        if let ConstValue::Unevaluated(def_id, substs) = constant.val {
+    fn compute_array_len(&mut self, constant: ty::LazyConst<'tcx>) {
+        if let ty::LazyConst::Unevaluated(def_id, substs) = constant {
```

Comment:
Why was this change made? If I called something `const` it's because it's meant to be used with const generics in the future.
This needs to be renamed back, and used for all `ty::Const`s in all `Substs`.
The lack of WF const handling probably allows a lot of unintended const expressions.
cc @varkor @yodaldevoid